src/routes/player/PlayQuiz.tsx

```diff
@@ -42,58 +44,103 @@ const PlayQuiz = observer(() => {
     }
   });
 
-// show answer
-useObserveEffect(() => {
-  const handleShowAnswer = (data: { correctAnswer: string }) => {
-    stopCountdown();
-    showTimer.set(false);
-    console.log(data);
-    const isCorrect = state$.quiz.selectedAnswer.get() === data.correctAnswer;
-
-    state$.quiz.isAnswerCorrect.set(isCorrect);
-    state$.quiz.displayQuestion.set(false);
-    state$.quiz.waitingForOthers.set(false);
-    state$.quiz.answered.set(true);
-    console.log('showing answer');
-  };
-
-  socket.on(WS_EVENTS.SHOW_ANSWER, handleShowAnswer);
-
-  return () => {
-    socket.off(WS_EVENTS.SHOW_ANSWER, handleShowAnswer);
-  };
-});
+  // show answer
+  useObserveEffect(() => {
+    const handleShowAnswer = (data: { correctAnswer: string }) => {
+      const isCorrect = state$.quiz.selectedAnswer.get() === data.correctAnswer;
 
-return (
-  <div className="bg-gray-100 min-h-screen flex flex-col items-center gap-5">
-    <div className="bg-purple-800 min-w-full p-5">
-      <Link to="/" className="text-6xl text-white font-bold">
-        !Toohak
-      </Link>
-    </div>
-    <div className="flex w-full p-2 justify-end">
-      <p className="text-lg">ID: {state$.player.id.get()}</p>
-    </div>
-    <div className="flex flex-col gap-6">
-      {showTimer.get() && <div className="text-2xl text-center">
-        <span className="font-semibold">Time Left:</span>&nbsp;{count} seconds
-      </div>}
-      {state$.quiz.displayQuestion.get() && <DisplayQuestion />}
-      {state$.quiz.waitingForOthers.get() && <WaitingForOtherPlayers />}
-      {
-        state$.quiz.answered.get()
-        && !state$.quiz.waitingForOthers.get()
-        && <ShowAnswer />
-      }
-      {
-        !state$.quiz.displayQuestion.get()
-        && !state$.quiz.waitingForOthers.get()
-        && !state$.quiz.answered.get()
-        && <WaitingForQuestion />
+      // stop, reset & hide timer
+      stopCountdown();
+      resetCountdown();
+      showTimer.set(false);
+
+      // update state
+      state$.quiz.isAnswerCorrect.set(isCorrect);
+      state$.quiz.displayQuestion.set(false);
+      state$.quiz.waitingForOthers.set(false);
+      state$.quiz.answered.set(true);
+
+      // emit waiting for quiz event after 5 seconds of showing answer
+      const waitForQuizTimeout = setTimeout(() => {
+        console.log('waiting for quiz');
+        socket.emit(WS_EVENTS.WAIT_FOR_QUIZ, state$.roomId.get());
```

Comment:
can you also include the player id so that I can return the player score in the backend

Reply:
not a good idea to do that rn, cuz the current version of the BE is doesn't accept a `playerId`. We can add it in during your change